src/neo/SmartContract/Native/DesignateContract.cs

```diff
@@ -0,0 +1,73 @@
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.SmartContract.Manifest;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Neo.SmartContract.Native
+{
+    [Flags]
+    public enum Role : byte
+    {
+        StateValidator = 0b00000001,
+        Oracle = 0b00000010
+    }
+
+    public sealed class DesignateContract : NativeContract
```

Comment:
`AccessControlContract`?

Reply:
Why call `AccessControl`?
This is a contract that committee designate some nodes as some role.